test/stripe/account_test.rb

```diff
@@ -2,20 +2,35 @@
 
 module Stripe
   class AccountTest < Test::Unit::TestCase
-    should "be retrievable" do
-      resp = make_account({
-        :charges_enabled => false,
-        :details_submitted => false,
-        :email => "test+bindings@stripe.com",
-      })
-      @mock.expects(:get).
-        once.
-        with('https://api.stripe.com/v1/account', nil, nil).
-        returns(make_response(resp))
-      a = Stripe::Account.retrieve
-      assert_equal "test+bindings@stripe.com", a.email
-      assert !a.charges_enabled
-      assert !a.details_submitted
+    should "be retrievable with generated responses" do
+      without_legacy_stubs do
+        stub_api do
+          get "/v1/account" do
+            generated_response.merge!({
+              :charges_enabled => false,
+              :details_submitted => false,
+              :email => "test+bindings@stripe.com",
+            })
+          end
+        end
+
+        a = Stripe::Account.retrieve
+        assert_equal "test+bindings@stripe.com", a.email
+        assert !a.charges_enabled
+        assert !a.details_submitted
+      end
+    end
+
+    # This test is made to demonstrate that we could offer a potential route
+    # for very easy-to-build tests that provide a very basic check that a
+    # method hits an API endpoint and responds with something. The idea here is
+    # that it could help our situation where tests are hard to write
+    # (especially for new resources that need new `TestData`, and are therefore
+    # not written).
+    should "be retrievable with generated responses (easy edition)" do
+      without_legacy_stubs do
+        Stripe::Account.retrieve
```

Comment:
@kyleconroy Cool! I took one more look at this and came up with a fourth option that's pretty lightweight and minimizes new code by making Webmock work a bit harder. Here's the diff (it's also pushed to this branch):

``` patch
diff --git a/test/stripe/account_test.rb b/test/stripe/account_test.rb
index 80bfbec..f9647b0 100644
--- a/test/stripe/account_test.rb
+++ b/test/stripe/account_test.rb
@@ -18,6 +18,8 @@ module Stripe
         assert_equal "test+bindings@stripe.com", a.email
         assert !a.charges_enabled
         assert !a.details_submitted
+
+        assert_requested :get, "#{Stripe.api_url}/v1/account"
       end
     end

@@ -30,6 +32,7 @@ module Stripe
     should "be retrievable with generated responses (easy edition)" do
       without_legacy_stubs do
         Stripe::Account.retrieve
+        assert_requested :get, "#{Stripe.api_url}/v1/account"
       end
     end
```

As you can see, it minimizes new code, and also allows us to assemble very lightweight (two line) endpoint tests that provide high-quality responses while also guaranteeing that endpoints were called.

The only downside is that the framework doesn't force you to assert the right URL, but the current state of affairs doesn't provide this guarantee either so we're not losing anything.

Thoughts?